api/py/ai/chronon/repo/validator.py

```diff
@@ -110,7 +109,29 @@ def get_pre_derived_join_columns(join: Join) -> List[str]:
             prefix = jp.prefix + "_" if jp.prefix else ""
             gb_prefix = jp.groupBy.metaData.name.replace(".", "_")
             output_columns.append(prefix + gb_prefix + "_" + col)
-    return output_columns
+    return output_columns + get_external_columns(join)
```

Comment:
Do we ever print out all available column names when there is a validation failure? It would be nice to have it but doesn't have to be in this PR